CONTRIBUTING.md

```diff
@@ -24,70 +42,46 @@ contributing enhancements to the Loki charm.
-    juju create-storage-pool operator-storage kubernetes storage-class=microk8s-hostpath
-```
-
-
-## Developing
-
-Create and activate a virtualenv with the development requirements:
-
-```bash
-   virtualenv -p python3 venv
-   source venv/bin/activate
-   pip install -r requirements.txt
-```
-
-
-Later on, upgrade packages as needed
-
-```bash
-   pip install --upgrade -r requirements.txt
-```
-
+- To handle our virtualenvs we use [`tox`](https://tox.wiki/en/latest/#), so there is no need
+  to deal with them manually. Tox will create, update, activate and deactivate virtualenvs for us.
 
 ### Testing
+
 All tests can be executed by running `tox` without arguments.
 
-To run individual test environments:
+Besides you can run individual test environments or tasks:
 
-```shell
-tox -e prettify  # update your code according to linting rules
-tox -e lint  # check your code complies to linting rules
-tox -e static # run static analysis
-tox -e unit  # run unit tests
+```bash
+tox -e lint         # check your code complies to linting rules
+tox -e static       # run static analysis
+tox -e unit         # run unit tests
+tox -e integration  # run integration tests
+tox -e fmt          # update your code according to linting rules
 ```
 
 Unit tests are implemented using the Operator Framework test [harness](https://ops.readthedocs.io/en/latest/#module-ops.testing).
 
 ### Build
 
-Install the [charmcraft tool](https://juju.is/docs/sdk/setting-up-charmcraft) and build the charm in this git repository:
+In order to build the charm so it can be deployed in [MicroK8s](https://microk8s.io/) using [Juju](https://juju.is/), we use [charmcraft](https://juju.is/docs/sdk/setting-up-charmcraft).
+So in the charm repository you have to run:
 
 ```bash
-    charmcraft pack
+$ charmcraft pack
+Packing charm 'loki-k8s_ubuntu-20.04-amd64.charm'...
+Created 'loki-k8s_ubuntu-20.04-amd64.charm'.
 ```
 
 ## Code Overview
 
-The core implementation of this charm is represented by the [`LokiOperatorCharm`](src/charm.py) class.
-`LokiOperatorCharm` responds to the following events:
+The core implementation of this charm is represented by the [`LokiOperatorCharm`](src/charm.py) class, which
+responds to the following events:
 
+- `install`: Here we patch k8s service.
+- `config_change`: Here we configure the charm.
 - `loki_pebble_ready`: Here we set up pebble layer and start the service
-- `relation_joined` In this event (Provided by the object `LokiPushApiProvider`) we set the `loki_push_api` (`http://{self.unit_ip}:{self.charm.port}/loki/api/v1/push`) so it can be used by a Consumer charm that uses the `LokiPushApiConsumer` object.
-
-Both clases `LokiPushApiProvider` and `LokiPushApiConsumer` are provided by the [`Loki library`](lib/charms/loki_k8s/v0/loki.py)
+- `upgrade_charm`: Here we patch k8s service and configure the charm.
+- `alertmanager_consumer.on.cluster_changed`: This event is provided by the AlertmanagerConsumer object. Here we configure the charm.
```

Comment:
```suggestion
- `alertmanager_consumer.on.cluster_changed`: This event is provided by the AlertmanagerConsumer object. Here we configure the charm.

### Library code
TODO
```

Reply:
The Loki libraries' code is explained in INTEGRATING.md file

Reply:
The suffix "object" adds no additional value. See suggestion above.